test/maxtree.jl

```diff
@@ -105,6 +105,12 @@ bbox2d(x1, y1, x2, y2) = (CI2(x1, y1), CI2(x2, y2))
     mtree101 = MaxTree(A101, connectivity=2, rev=false)
     @test mtree101 isa MaxTree{3}
     @test diameters(mtree101) == Array{Int}(undef, 1, 0, 1)
+
+    # test offset arrays
+    Am11 = OffsetArray(A, -1, -1)
+    am11_tree = MaxTree(Am11, connectivity=1, rev=false)
+    @test size(am11_tree) == size(Am11)
```

Comment:
As I've said, this should be `@test axes(am11_tree) == axes(Am11)` and the test can't guarantee that `Am11[idx]` and `am11_tree.parentindices[idx]` points to the same pixel.

Reply:
> As I've said, this should be `@test axes(am11_tree) == axes(Am11)`

After dropping `A` type param, `MaxTree` doesn't store the axes of the original image. But we don't really need them, since it's not required to produce the correct results.
By correct results I mean tests like this: `@test collect(area_opening(Am11)) == area_opening(A)`.
But, as an additional bonus (since we use `similar(image)` to initialize output), the operations preserve the axes of `image`, I've just updated the tests to check that.

> and the test can't guarantee that Am11[idx] and am11_tree.parentindices[idx] points to the same pixel.

The MaxTree-based operations only use linear indices as `idx`. So far it looks like this assumption is true.

Reply:
> The MaxTree-based operations only use linear indices as idx. So far it looks like this assumption is true.

Right, thanks for the explanation!